hydra-node/src/Hydra/Chain/CardanoClient.hs

```diff
@@ -245,69 +273,92 @@ queryEpochNo networkId socket queryPoint = do
 --
 -- Throws at least 'QueryException' if query fails.
 queryProtocolParameters ::
+  -- | Current network discriminant
   NetworkId ->
+  -- | Filepath to the cardano-node's domain socket
   SocketPath ->
   QueryPoint ->
   IO (PParams LedgerEra)
 queryProtocolParameters networkId socket queryPoint = do
-  let query =
-        QueryInEra
-          BabbageEraInCardanoMode
-          ( QueryInShelleyBasedEra
-              ShelleyBasedEraBabbage
-              QueryProtocolParameters
-          )
-  runQuery networkId socket queryPoint query >>= throwOnEraMismatch
+  AnyCardanoEra era <- queryCurrentEra networkId socket QueryTip
+  mkQueryInEra era QueryProtocolParameters
+    >>= runQuery networkId socket queryPoint
+    >>= throwOnEraMismatch
+    >>= coercePParamsToLedgerEra era
+ where
+  encodeToEra eraToEncode pparams =
+    case eitherDecode' (encode pparams) of
+      Left e -> throwIO $ QueryProtocolParamsEncodingFailureOnEra (anyCardanoEra eraToEncode) (Text.pack e)
+      Right (ok :: PParams LedgerEra) -> pure ok
+
+  coercePParamsToLedgerEra :: CardanoEra era -> PParams (ShelleyLedgerEra era) -> IO (PParams LedgerEra)
+  coercePParamsToLedgerEra era pparams =
+    case era of
+      ByronEra -> throwIO $ QueryProtocolParamsEraNotSupported (anyCardanoEra ByronEra)
+      ShelleyEra -> encodeToEra ShelleyEra pparams
+      AllegraEra -> encodeToEra AllegraEra pparams
+      MaryEra -> encodeToEra MaryEra pparams
+      AlonzoEra -> encodeToEra AlonzoEra pparams
+      BabbageEra -> pure pparams
+      ConwayEra -> encodeToEra ConwayEra pparams
 
 -- | Query 'GenesisParameters' at a given point.
 --
 -- Throws at least 'QueryException' if query fails.
-queryGenesisParameters :: NetworkId -> SocketPath -> QueryPoint -> IO (GenesisParameters ShelleyEra)
-queryGenesisParameters networkId socket queryPoint =
-  let query =
-        QueryInEra
-          BabbageEraInCardanoMode
-          ( QueryInShelleyBasedEra
-              ShelleyBasedEraBabbage
-              QueryGenesisParameters
-          )
-   in runQuery networkId socket queryPoint query >>= throwOnEraMismatch
+queryGenesisParameters ::
+  -- | Current network discriminant
+  NetworkId ->
+  -- | Filepath to the cardano-node's domain socket
+  SocketPath ->
+  QueryPoint ->
+  -- | The current running era we can use to query the node
+  CardanoEra era ->
```

Comment:
Could: Make this consistent with other era-specific queries and do it in "inside" with `queryCurrentEraExpr`